### PR TITLE
Make bubble sort and selection sort optional lessons.

### PR DESCRIPTION
**Pull request by alope107**
The two sorting algorithms made optional are not often seen in practice. They are also not really asked about in interviews anymore. Even without them, this module is still quite large and abstract. These changes will hopefully free up students to devote their time elsewhere if they need, while still allowing those who are interested to explore.

See preview here: https://learn-2.galvanize.com/cohorts/3027

### CODE REVIEW

File: sorting-algorithms/insertion-sort.md [hunk 1]
```diff
@@ -72,17 +83,35 @@ The insertion sort algorithm requires (in the worst case):
...
-We might also notice that insertion sort, like bubble sort, has a best case time complexity of _O(n)_ if the list is already sorted. As we expand the sorted sub-part of the array, each new item is already in its correct spot, which we can determine with a single check per item. Since there are _n_ items, this gives a best case of _O(n)_.
+### !callout-info
+
+## Mathematical Proofs Out of Scope
+
+In general, full mathematical proofs for the algorithms we discuss are out of scope for the curriculum. We will often apply analyses that are more intuitive than rigorous. However, there are great resources online that explain the math in-depth! Follow your curiosity!
+
+### !end-callout
+
+Therefore insertion sort has a time complexity of _O(n<sup>2</sup>)_.
```
Comment (anselrognlie) [inline sorting-algorithms/insertion-sort.md:102]: Can we move this line above the callout? It feels like this got cut off from the preceding argument.

File: sorting-algorithms/problem-set-sorting-algorithms.md [hunk 2]
```diff
@@ -212,7 +31,7 @@ The insertion sort algorithm is unique because...
 ### !end-challenge
 <!-- prettier-ignore-end -->
 
-<!-- Question 7 -->
+<!-- Question 2 -->
```
Comment (anselrognlie) [inline sorting-algorithms/problem-set-sorting-algorithms.md:34]: I'm not a fan of having the question number comments in the markdown. I know this is maintaining the consistency with what's here already, but these just seem to be potential busy work.

**Review by CheezItMan — APPROVED** [review]
I like it, it brings down the number of things students have to conceptualize and focuses more on the essential concepts.  I also like the more grounded explanations of the time complexities and discussion about how to pick an algorithm.

File: sorting-algorithms/problem-set-sorting-algorithms.md [hunk 3]
```diff
@@ -272,7 +67,7 @@ For algorithms with equal time complexity, order them in alphabetical order.
 
 There's nothing like getting familiar with sorting algorithms than to implement it yourself!
```
Comment (anselrognlie) [inline sorting-algorithms/problem-set-sorting-algorithms.md:68]: Outside your change, but this jumped out me as being really strangely worded. Could we slip this in?

```suggestion
There's no better way to get familiar with sorting algorithms than to implement one ourselves!
```

**Review by anselrognlie — APPROVED** [review]
Just a couple minor things at your discretion. Looks good. Thanks for the revisions!

**Pull request merged**